tls/s2n_ktls.c

```diff
@@ -86,9 +87,95 @@ S2N_RESULT s2n_ktls_retrieve_file_descriptor(struct s2n_connection *conn, s2n_kt
     return S2N_RESULT_OK;
 }
 
+S2N_RESULT s2n_ktls_init_aes128_gcm_crypto_info(struct s2n_connection *conn, s2n_ktls_mode ktls_mode,
+        struct s2n_key_material *key_material, struct s2n_tls12_crypto_info_aes_gcm_128 *crypto_info, int *tls_tx_rx_mode)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(conn->client);
+    RESULT_ENSURE_REF(conn->server);
+    RESULT_ENSURE_REF(key_material);
+    RESULT_ENSURE_REF(crypto_info);
+    RESULT_ENSURE_REF(tls_tx_rx_mode);
+    RESULT_GUARD(s2n_ktls_validate(conn, ktls_mode));
+
+    /* TODO once other ciphers and protocols are supported, check the
+     * negotiated cipher (AES_128_GCM). This would involve adding a unique
+     * identifier to s2n_cipher.
+     */
+
+    /* set tls mode */
+    if (ktls_mode == S2N_KTLS_MODE_SEND) { /* TX */
+        *tls_tx_rx_mode = S2N_TLS_TX;
+    } else { /* RX */
+        *tls_tx_rx_mode = S2N_TLS_RX;
+    }
```

Comment:
Why is this part of s2n_ktls_init_aes128_gcm_crypto_info?

Reply:
I can rename the fn but its init-ing all values needed to enable ktls on the socket.

Reply:
But why do that all in one method? It makes the purpose of the method less obvious, and you'll notice also complicates the signature. Separate methods will be clearer.